met-web/src/components/engagement/form/EngagementWidgets/Documents/DocumentFolder.tsx

```diff
@@ -1,26 +1,126 @@
-import React from 'react';
-import { Grid, IconButton, Stack, Typography } from '@mui/material';
+import React, { useState, useEffect, useContext } from 'react';
+import { Grid, IconButton, Stack, TextField, Typography } from '@mui/material';
 import { MetWidgetPaper } from 'components/common';
 import { DocumentItem } from 'models/document';
 import FolderIcon from '@mui/icons-material/Folder';
 import HighlightOffIcon from '@mui/icons-material/HighlightOff';
-import { When } from 'react-if';
+import { If, Then, Else, When } from 'react-if';
 import SubdirectoryArrowRightIcon from '@mui/icons-material/SubdirectoryArrowRight';
 import DocumentSwitch from './DocumentSwitch';
+import { useAppDispatch } from 'hooks';
+import { openNotificationModal } from 'services/notificationModalService/notificationModalSlice';
+import { deleteDocument, patchDocument, PatchDocumentRequest } from 'services/widgetService/DocumentService.tsx';
+import { WidgetDrawerContext } from '../WidgetDrawerContext';
+import { WidgetType, Widget } from 'models/widget';
+import Edit from '@mui/icons-material/Edit';
+import { DocumentsContext } from './DocumentsContext';
+import { updatedDiff } from 'deep-object-diff';
+import { openNotification } from 'services/notificationService/notificationSlice';
 
 const DocumentFolder = ({ documentItem }: { documentItem: DocumentItem }) => {
+    const dispatch = useAppDispatch();
+    const { documents, loadDocuments } = useContext(DocumentsContext);
+    const { widgets, loadWidgets } = useContext(WidgetDrawerContext);
+    const documentWidget = widgets.find((widget: Widget) => widget.widget_type_id === WidgetType.Document);
+    const [edit, setEdit] = useState<boolean>(false);
+    const [document, setDocument] = useState<DocumentItem | undefined>(documentItem);
+
+    useEffect(() => {
+        const updatedDocument = documents.find((document: DocumentItem) => document.id === documentItem.id);
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // ts-ignore:
```

Comment:
what is being ignored here?

Reply:
I thought i removed that sorry about that that's weird

Reply:
I thought i removed that sorry about that that's weird removed.